ci/templates/ci/dashboard.html

```diff
@@ -176,7 +184,11 @@ <h2 class="dashhead-title">{{ dashboard.name }} Dashboard</h2>
     metrics(widget)
       .forEach(function(d) {
         if (widget['type'] == 'pie'){
-          d.value = data[d.key][data[d.key].length-1].y;
+          if (data[d.key].length == 0) {
```

Comment:
you can simplify this by using the shorthand:
`d.value = data[d.key].length > 0 ? data[d.key][data[d.key].length-1].y : 0;`